orm/src/main/java/org/hibernate/search/query/hibernate/impl/FullTextQueryImpl.java

```diff
@@ -289,11 +290,20 @@ public FullTextQuery setResultTransformer(ResultTransformer transformer) {
 		return this;
 	}
 
+	/*
+	 * Implementation note: this method is defined as generic in the interface,
+	 * but we must implement it without generics (otherwise it won't compile).
+	 *
+	 * The actual reason is a bit hard to explain: basically we implement
+	 * javax.persistence.Query as a raw type at some point, and our superclass
+	 * (also extended as a raw type) also implements this interface, but as a non-raw type.
+	 * This seems to confuse the compiler, which thinks there are two different methods.
+	 */
```

Comment:
I've been trying to understand this but couldn't figure out why this is necessary :-/

@emmanuelbernard @hferentschik @gunnarmorling ? You all are the generics gurus..

In case we have to integrate this, will it cause runtime compatbility issues with our previous signature?
(I think not, as our API on the interface is still the generic one... but that just makes it even more odd)